watttime/tcy.py

```diff
@@ -0,0 +1,109 @@
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import pandas as pd
+import pytz
+from typing import Optional, List
+from .api import WattTimeHistorical
+
+@dataclass
+class TCYConfig:
+    """Configuration for TCY calculation"""
+    region: str
+    timezone: str
+    holidays: Optional[List[str]] = None
+
+class TCYCalculator:
+    """Calculate Typical Carbon Year profiles from historical MOER data using a 3-year lookback period"""
+    
+    def __init__(self, username: str, password: str, config: TCYConfig):
+        self.wt_client = WattTimeHistorical(username, password)
```

Comment:
can this class just be a child of WattTimeHistorical?

Reply:
I had suggested making this independent, since `WattTimeHistorical` is a child of `WattTimeAPI` and this isn't really an API object, but rather a utility calculation that uses API data.